CONTRIBUTING.md

```diff
@@ -134,15 +134,33 @@ We follow [youtube-dl's policy](https://github.com/ytdl-org/youtube-dl#can-you-a
 
 # DEVELOPER INSTRUCTIONS
 
-Most users do not need to build yt-dlp and can [download the builds](https://github.com/yt-dlp/yt-dlp/releases) or get them via [the other installation methods](README.md#installation).
+Most users do not need to build yt-dlp and can [download the builds](https://github.com/yt-dlp/yt-dlp/releases), get them via [the other installation methods](README.md#installation) or directly run it using `python -m yt_dlp`.
 
-To run yt-dlp as a developer, you don't need to build anything either. Simply execute
+`yt-dlp` uses [`hatch`](<https://hatch.pypa.io>) as a project management tool.
+You can easily install it using [`pipx`](<https://pipx.pypa.io>) via `pipx install hatch`, or else via `pip` or your package manager of choice.
 
-    python -m yt_dlp
+If you plan on contributing to `yt-dlp`, you are required to run
 
-To run all the available core tests, use:
+```shell
+$ hatch run install
+```
+
+before creating commits so that the pre-commit hooks are properly initialized.
+After this you can use `hatch shell` to enable a virtual environment that has development dependencies as well as `yt-dlp` installed.
+
+Scripts can be used to run simple tasks, like linting or testing, without having to run `hatch shell` first:
 
-    python devscripts/run_tests.py
+```shell
+$ hatch run format
+$ hatch run lint
```

Comment:
`hatch run fix` runs both now. Im debating if this code block should be included at all since it makes it seem like a list of commands to run; instead it should just show what important `hatch run` commands exist